### ISSUE TITLE
Change requested for AppCenter: binary uploads are not allowed

### ISSUE THREAD

**Issue by cassidyjames** [writeas/writeas-gtk#16]
While doing our routine review of Write.as for AppCenter, we noticed that you've included the CLI tool as a binary in your repo and are installing it with meson. However, for transparency reasons binaries are not allowed to be submitted to AppCenter; all binaries must be built on the AppCenter infrastructure to ensure reproducible builds and auditability.

This is briefly mentioned in our [Technical Requirements](https://github.com/elementary/houston/wiki/Before-You-Publish#technical-requirements) for publishing apps. We have since elaborated on this requirement: 

>To ensure reproducible builds, transparency, and auditability, binaries cannot be uploaded or included alongside the source code to be installed on users' devices.

Your app is super close to being able to be shipped to users, but we'll need to solve this somehow in a different way. :confused:

**Comment by thebaer** [comment]
Is there a quick way I can test this requirement with Houston CI? Or will it catch this issue when I push up a release with the CLI build process included?

My ideas is to try including the [CLI](https://github.com/writeas/writeas-cli) as a git submodule and then following [this tutorial](https://github.com/heisantosh/howto-golang-gtk) to get it to build with Meson. Do you think this is a decent route, or do you know of a better way?

**Comment by cassidyjames** [comment]
The submodule route sounds sane to me! You might poke one of the other developers of Go apps for elementary OS for pointers, as well. I will try and grab some links to examples. I'm not sure the best way to test, other than testing locally and using Houston CI on GitHub. You could also try submitting your app and checking the build directory at http://packages.elementary.io/testing/pool/main/c/com.github.writeas.writeas-gtk/ for the built version and testing that on a clean elementary OS install, i.e. in a VM.

**Comment by thebaer** [comment]
Great :+1: I'll give that a shot.

**Comment by cassidyjames** [comment]
@thebaer here are a couple of apps built for elementary OS with Go, if it helps:

https://github.com/rkoesters/xkcd-gtk

https://github.com/heisantosh/qrshare

**Comment by thebaer** [comment]
Thanks! I'm taking what I can from those.

By the way, does the built CLI need to be named with RDNN? Or can I have it installed as `writeas`?

**Comment by thebaer** [comment]
Closed by #17

**Comment by cassidyjames** [comment]
@thebaer did you mean to embed the source wholesale versus using a submodule?

**Comment by thebaer** [comment]
I included the base CLI as a submodule, but embedded its dependencies on purpose, yes. That's because the `gb` tool I'm using for the build process doesn't support importing libraries from non-GitHub URLs, of which the CLI has a few. So I just embedded them instead of creating new forks of those libraries on a github.com-based repo, updating the CLI import paths, and then having the submodule point to that special CLI branch.

Just wanted to get everything built. I'll clean this up / find a better solution in the future.
